### DISCUSSION TITLE
widget feature | filter events, products and variations at the same time

### DISCUSSION THREAD

**Discussion by alemao8** [pretix/pretix#3159] [Q&A]
Hello Pretix team!

I would like to filter some events, products and variations at the same time using the widget feature.
I have x events, y products and z variations in some products. So I would like to filter and display in one calendar a part of x events, a part of y products of each event and a part of z variations of products.
Is this possible ?
I tried using theorganizer link and then filtering events, items and variations as mentioned in https://docs.pretix.eu/en/latest/user/events/widget.html#filtering-products but I cannot combine filtering the items and variations **at the same time.**

Thank you in advance

**Comment by wiffbi** [comment]
AFAIK you cannot filter events in the calendar-overview – it is either one specific event in the event-attribute URL or all events. The HTML-attributes `items`, `categories` and `variations` only work on event-level when a product-list is shown – the list of products will be filtered. You can combine those filters though by providing all or some of those attributes. You shoudl even be able to pass those filters to the widget when using the organizer-URL to show all events and then after selecting an event, the products should be filtered according to those HTML-attributes.

**Reply by alemao8** [reply]
Thanx for your reply wiffbi.
Actually I managed to filter a number of events in the calendar overview. I used the organizer URL and then displayed some events by using item metadata.
What I am missing for the rest of my case, is whether I can filter specific variations after filtering specific products. Also what are the operators to combine the above. AND/NOT etc.

**Reply by wiffbi** [reply]
When you use the filter-attribute to filter events by event meta-data, you can filter by multiple meta-data-values by creating an url-encoded string like so:

`<pretix-widget event="https://pretix.eu/demo/series/" list-type="list" filter="attr[Promoted]=Yes&attr[Highlight]=Yes"></pretix-widget>`

These meta-data-values in filter are always combined through OR.

Regarding filtering products, you would just simply add the filters you need, like so:

```
<pretix-widget
    event="https://pretix.eu/demo/series/"
    list-type="list"
    filter="attr[Promoted]=Yes&attr[Highlight]=Yes"
    items="2,5,9"
    variations="23,40,90"
></pretix-widget>
```
Note though, that the HTML attributes `items`, `categories` and `variations` are always combined with AND. So if an item from the items-HTML-attribute does not have a variation listed in the variations-HTML-attribute, the whole item will be filtered. So basically once you add a variations-filter, all the items you filtered need a variation, which needs to be listed in the variations-HTML-attribute otherwise the item will be filtered.
